### PR TITLE
wiimms-szs-tools (new formula)

### PR DESCRIPTION
**Pull request by larsenv**
Wiimms SZS Tools is a library to read/write from a lot of file formats used especially in Wii games, especially Mario Kart Wii.

- [X] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [X] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [X] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)? **(yes, but one exception)**

-----

The only problem that brew audit returns is something out of my control - it says that the GitHub repository is not popular enough.

I will however say that the GitHub repository is new. Wiimms SZS Tools has been around for years and is a popular and very-well-made tool used in the Wii game modding community. The source code was originally in a SVN repository, then as source tarballs on his site, then made into a GitHub repository so he can use a buildbot to compile the app as the developer who made these tools doesn't have access to a Mac.

I hope that the formula can be added, because I worked hard on creating my first formula and I've wanted this tool to be a package installable with brew for a long time.

### CODE REVIEW

**Comment by dawidd6** [comment]
```
wiimms-szs-tools:
  * GitHub repository not notable enough (<30 forks, <30 watchers and <75 stars)
```

**Comment by larsenv** [comment]
The build is currently failing, I'm not sure how I'm supposed to fix it? It worked for me on my machine.

File: Formula/wiimms-szs-tools.rb
Comment (harens) [inline Formula/wiimms-szs-tools.rb:4]:
```suggestion
  homepage "https://szs.wiimm.de/"
  sha256 "db8b73e41c2bc80e71f6a8fc95513f9a4424800e55dbcfcf1ea4f103b6c61265"
  url "https://download.wiimm.de/source/wiimms-szs-tools/wiimms-szs-tools.source-2.19a.txz"
```

Just looking throught the logs, it states the following:
```
A checksum was not provided for this resource.
For your reference the SHA-256 is: db8b73e41c2bc80e71f6a8fc95513f9a4424800e55dbcfcf1ea4f103b6c61265
```

Hopefully this should help.

**Review by harens — CHANGES_REQUESTED** [review]

File: Formula/wiimms-szs-tools.rb
Comment (SMillerDev) [inline Formula/wiimms-szs-tools.rb:27]: We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

File: Formula/wiimms-szs-tools.rb
Comment (SMillerDev) [inline Formula/wiimms-szs-tools.rb:15]: This would be better as a patch upstream to allow `DESTDIR` to override this. It's how most other projects work with make.

File: Formula/wiimms-szs-tools.rb
Comment (SMillerDev) [inline Formula/wiimms-szs-tools.rb:22]: Is there no `make install`?

Comment (larsenv) [reply Formula/wiimms-szs-tools.rb:23]: There is, but I think it requires sudo and idk how to get around that.

File: Formula/wiimms-szs-tools.rb
Comment (larsenv) [reply Formula/wiimms-szs-tools.rb:16]: I'm not sure how to do that.

File: Formula/wiimms-szs-tools.rb
Comment (larsenv) [reply Formula/wiimms-szs-tools.rb]: I just changed it to create an empty archive, hope that's better.

File: Formula/wiimms-szs-tools.rb
Comment (gromgit) [reply Formula/wiimms-szs-tools.rb:16]: I think @SMillerDev is saying to raise this issue with the upstream project, so that it gets fixed for everyone.

Comment (larsenv) [reply Formula/wiimms-szs-tools.rb:16]: I don't think the guy who owns the GitHub repository uses its features to be honest, it's kind of introverted. He's also a busy person. I don't think he would see a PR, likewise merge it.

Comment (SMillerDev) [reply Formula/wiimms-szs-tools.rb:16]: That would be a bigger blocker for me than the notability. We need to be able to rely on software being maintained and supported to ship it in homebrew. https://docs.brew.sh/Acceptable-Formulae#niche-or-self-submitted-stuff

Comment (larsenv) [reply Formula/wiimms-szs-tools.rb:16]: Well the software is maintained, but I think the guy mostly keeps to himself on this project.

Comment (SMillerDev) [reply Formula/wiimms-szs-tools.rb:16]: If they don't accept contributions I wouldn't really consider it open source. Either way, it should be submitted upstream or Homebrew can't merge it.

Comment (larsenv) [reply Formula/wiimms-szs-tools.rb:16]: Sorry to ask but how do I make it upstream? Do I make a fork of it and then modify that and then modify the formula to use my fork?

Comment (SMillerDev) [reply Formula/wiimms-szs-tools.rb:16]: You make a fork and then you make a pull request to the original repository to include your changes. Then you point the formula to the new release that includes your changes.

Comment (larsenv) [reply Formula/wiimms-szs-tools.rb:16]: OK. The maintainer added some more commits to the repository that have to do with Mac builds and now compiling results in an error. For now I'll open an issue.

Comment (larsenv) [reply Formula/wiimms-szs-tools.rb:16]: https://github.com/Wiimm/wiimms-szs-tools/issues/1

Comment (larsenv) [reply Formula/wiimms-szs-tools.rb:16]: https://github.com/Wiimm/wiimms-szs-tools/issues/1

**Comment by Leseratte10** [comment]
I just saw this issue being linked from Wiimms repository. I know that Wiimm wanted to fix some more Mac stuff in the near future, maybe this is one of the things. I'm not sure if he's seen the issue in the repo yet. When I next talk to him I'll ask him about adding a parameter to override that destination directory. 

And I noticed another thing: The formula in this PR states `license "GPL-2.0-or-later"` which I believe is wrong. Where did you get the information that the SZS tools are under the GPL2+? Both the source code archive and the github repository contain a "gpl-2.0.txt" file, which does not seem to contain the usual "or any later version" statement. I believe the tool is under a GPL2-only license. If needed I can ask Wiimm about that as well.

**Comment by gromgit** [comment]
> Where did you get the information that the SZS tools are under the GPL2+?

From the [README](https://github.com/Wiimm/wiimms-szs-tools/blob/master/README.md):
> This program is free software; you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation; either version 2 of the License, **or (at your option) any later version**.

**Comment by Leseratte10** [comment]
Dammit. I should have checked that. I'll ask him to update / clarify the gpl-2.0.txt file as well then just to make it more clear.

**Comment by gromgit** [comment]
> I'll ask him to update / clarify the gpl-2.0.txt

From [gpl-2.0.txt](https://github.com/Wiimm/wiimms-szs-tools/blob/master/project/gpl-2.0.txt), highlights are mine:
> Everyone is permitted to copy and distribute verbatim copies of this license document, but **changing it is not allowed**.
> ...
> 	    How to Apply These Terms to Your New Programs
>
> To do so, attach the following notices **to the program**.  It is safest to attach them to the start of each source file to most effectively convey the exclusion of warranty; and each file should have at least the "copyright" line and a pointer to where the full notice is found.

Before changing a license file, it's wise to read through it first. As a general rule of thumb, changing _anything_ in license boilerplate, except the stuff explicitly tagged as modifiable, is usually the Wrong Thing to do.

**Comment by Leseratte10** [comment]
Yeah, I know that that's not supposed to be changed, but the file in the repo also contains the "how to apply these terms to your program" section which doesn't belong there and is confusing. I checked the source code files as well and these have the "default" copyright header with the "this is under the GPL either version 2 or any later version" stuff so GPL2-or-later seems to be correct. Sorry for the confusion.

**Comment by gromgit** [comment]
> the file in the repo also contains the "how to apply these terms to your program" section which doesn't belong there and is confusing

It _does_ belong there, because it's in the [original file from GNU](https://www.gnu.org/licenses/old-licenses/gpl-2.0-standalone.html), and the text I quoted above specifically says you're _not_ allowed to modify this file (that's what **verbatim copies** means).

**Comment by Leseratte10** [comment]
I thought the "END OF TERMS AND CONDITIONS" makes it clear that that's the end of the license and the bottom part is just additional information? https://opensource.stackexchange.com/questions/4350/applying-gnu-gpl-and-messages-printed-by-a-program 

Then again, it doesn't really hurt if that stays in there. The correct license is mentioned in the readme and in the code files, I was just looking at the wrong place.

**Comment by gromgit** [comment]
`END OF TERMS AND CONDITIONS` is just the matching "bracket" to `TERMS AND CONDITIONS FOR COPYING, DISTRIBUTION AND MODIFICATION` near the start. Just as you'd not remove anything before the latter, so there's no reason to remove any after the former, even though both chunks are "just additional information".

> it doesn't really hurt if that stays in there

Yes, and if you remove it...well, IANAL, but there may be a legal case for arguing that the license is no longer enforceable, because you changed the original text.

**Comment by iMichka** [comment]
Any new/progress here? When a new formula addition takes months, it's quite a bad sign: the maintainability is low, a lot of stuff had to be discussed, and the addition was not straightforward? This sounds like the quality of the formula does not meet or standards for addition.

**Comment by larsenv** [comment]
I'm considering closing this PR.

**Comment by iMichka** [comment]
Let's close this. You can still host the formula in a personal tap if needed.

**Comment by carlocab** [comment]
Here are some docs on creating on your own tap and building bottles on it:

https://docs.brew.sh/How-to-Create-and-Maintain-a-Tap
https://brew.sh/2020/11/18/homebrew-tap-with-bottles-uploaded-to-github-releases/

**Pull request closed without merging**
